src/components/jellyfinApi.ts

```diff
@@ -11,28 +11,53 @@ export abstract class JellyfinApi {
     // Address of server
     public static serverAddress: string | null = null;
 
+    // device name
+    public static deviceName = 'Google Cast';
+
+    // unique id
+    public static deviceId = '';
+
+    // version
+    public static versionNumber = RECEIVERVERSION;
+
     public static setServerInfo(
         userId: string,
         accessToken: string,
-        serverAddress: string
+        serverAddress: string,
+        receiverName: string
     ): void {
         console.debug(
-            `JellyfinApi.setServerInfo: user:${userId}, token:${accessToken}, server:${serverAddress}`
+            `JellyfinApi.setServerInfo: user:${userId}, token:${accessToken}, ` +
+                `server:${serverAddress}, name:${receiverName}`
         );
         this.userId = userId;
         this.accessToken = accessToken;
         this.serverAddress = serverAddress;
+
+        // remove special characters from the receiver name
+        receiverName = receiverName.replace(/[^\w\s]/gi, '');
+
+        if (receiverName) {
+            this.deviceName = receiverName;
+            // deviceId just needs to be unique-ish
+            this.deviceId = btoa(receiverName);
```

Comment:
This is nitpicky but why not generate an UUID or something else? a base64 representation of the device's name is not exactly an id.

We can also use ``sessionId`` from here: https://developers.google.com/cast/docs/reference/web_receiver/cast.framework.system.ApplicationData

Reply:
You're probably right.

This code has been refactored from maincontroller or stable, don't remember which exactly.

It fixes an error in execution order where nightly will register with the server before the device identity has been set, so after these properties are set it would identify itself differently and hence leave ghost entries in the server device list.

This is the part that I fixed, so my idea here is to just keep it as-is for now, and then we can try to change it and assert that it did not break later on.